### ISSUE TITLE
[CI] DynamicMappingIT.testMappingVersionAfterDynamicMappingUpdate failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#62339]
Looks like `DynamicMappingIT.testMappingVersionAfterDynamicMappingUpdate` is a newly added test that has already failed a number of times in CI.

**Build scan**:
https://gradle-enterprise.elastic.co/s/wa7vwdvq44qu4/tests/:server:internalClusterTest/org.elasticsearch.index.mapper.DynamicMappingIT/testMappingVersionAfterDynamicMappingUpdate#1

**Repro line**:
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.mapper.DynamicMappingIT.testMappingVersionAfterDynamicMappingUpdate" -Dtests.seed=610A20E1C684BE63 -Dtests.security.manager=true -Dtests.locale=ar-DZ -Dtests.timezone=America/Winnipeg -Druntime.java=11`

**Reproduces locally?**:
Did not reproduce for me.

**Applicable branches**:
`master`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.buildToolTypes=gradle&search.buildToolTypes=maven&search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.index.mapper.DynamicMappingIT&tests.sortField=FAILED&tests.test=testMappingVersionAfterDynamicMappingUpdate&tests.unstableOnly=true

**Failure excerpt**:
```
org.elasticsearch.index.mapper.DynamicMappingIT > testMappingVersionAfterDynamicMappingUpdate FAILED
    java.lang.AssertionError: 
    Expected: <2L>
         but: was <1L>
        at __randomizedtesting.SeedInfo.seed([610A20E1C684BE63:F42EC2B1B99E914A]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.mapper.DynamicMappingIT.testMappingVersionAfterDynamicMappingUpdate(DynamicMappingIT.java:168)
```

**Comment by mark-vieira** [comment]
@romseygeek do you mind investigating this given you authored this test?

**Comment by romseygeek** [comment]
This will be fixed by #62352

**Comment by cbuescher** [comment]
Looks like this can be closed since #62352 is merged
